api/bibliotheca.py

```diff
@@ -1301,8 +1267,11 @@ def __init__(self, _db, collection, api_class=BibliothecaAPI,
             self.api = api_class
         else:
             self.api = api_class(_db, collection)
+        self.replacement_policy = BibliothecaAPI.replacement_policy(
+            _db, self.analytics
+        )
         self.bibliographic_coverage_provider = BibliothecaBibliographicCoverageProvider(
-            collection, self.api
+            collection, self.api, replacement_policy=self.replacement_policy
```

Comment:
Propagating the replacement policy is what causes test_analytics_event to catch all the events.